src/psbt/mod.rs

```diff
@@ -334,31 +332,30 @@ impl<'psbt, Pk: MiniscriptKey + ToPublicKey> Satisfier<Pk> for PsbtInputSatisfie
             .map(|(pk, sig)| (*pk, *sig))
     }
 
-    fn check_after(&self, n: u32) -> bool {
-        let locktime = self.psbt.unsigned_tx.lock_time;
-        let seq = self.psbt.unsigned_tx.input[self.index].sequence;
-
-        // https://github.com/bitcoin/bips/blob/master/bip-0065.mediawiki
-        // fail if TxIn is finalized
-        if seq == 0xffffffff {
-            false
-        } else {
-            <dyn Satisfier<Pk>>::check_after(&After(locktime), n)
+    fn check_after(&self, n: LockTime) -> bool {
+        if !self.psbt.unsigned_tx.input[self.index].enables_lock_time() {
+            return false;
         }
+
+        let lock_time = LockTime::from(self.psbt.unsigned_tx.lock_time);
+
+        <dyn Satisfier<Pk>>::check_after(&lock_time, n)
     }
 
-    fn check_older(&self, n: u32) -> bool {
+    fn check_older(&self, n: Sequence) -> bool {
         let seq = self.psbt.unsigned_tx.input[self.index].sequence;
+
         // https://github.com/bitcoin/bips/blob/master/bip-0112.mediawiki
-        // Disable flag set. return true
-        if n & SEQUENCE_LOCKTIME_DISABLE_FLAG != 0 {
-            true
-        } else if self.psbt.unsigned_tx.version < 2 || (seq & SEQUENCE_LOCKTIME_DISABLE_FLAG != 0) {
-            // transaction version and sequence check
-            false
-        } else {
-            <dyn Satisfier<Pk>>::check_older(&Older(seq), n)
+        // Disable flag set => return true.
+        if !n.is_relative_lock_time() {
+            return true;
         }
+
+        if self.psbt.unsigned_tx.version < 2 || !seq.is_relative_lock_time() {
```

Comment:
I like the other mthods. They make the code here a lot cleaner

Reply:
Sorry mate I'm not following you, which other methods?

Reply:
Oops. I meant I really like the new `added` methods upstream. They make the code here a lot cleaner. There is nothing to resolve here

Reply:
:) Will be even better when `relative::LockTime` is available, its been nice to see the API in rust-bitcoin develop huh.